tools/codegen/gen_backend_stubs.py

```diff
@@ -256,7 +263,43 @@ def gen_dispatcher_registrations(
 
     assert class_name is not None
     backend_index = backend_indices[dispatch_key]
+
+    dispatch_registrations_body = list(concatMap(
+        dest.RegisterDispatchKey(
+            backend_index,
+            Target.REGISTRATION,
+            selector,
+            rocm=False,
+            cpp_namespace=cpp_namespace,
+            class_method_name=f'{class_name}'),
+        grouped_native_functions
+    ))
+    deferred_dispatch_registrations = ""
+    static_init_dispatch_registrations = ""
+    if eager_registration:
+        static_template = CodeTemplate("""\
+TORCH_LIBRARY_IMPL(aten, $dispatch_key, m) { 
+    $dispatch_registrations_body
+};""")
+        static_init_dispatch_registrations = static_template.substitute(
+            dispatch_key=dispatch_key,
+            dispatch_registrations_body=dispatch_registrations_body
+        )
+    else:
+        deferred_template = CodeTemplate("""\
+TORCH_API void Register${backend_name}${dispatch_key}NativeFunctions() { 
+    auto m = MAKE_TORCH_LIBRARY_IMPL(aten, $dispatch_key);
+    $dispatch_registrations_body
+}""")
+        deferred_dispatch_registrations = deferred_template.substitute(
+            backend_name=backend_name,
+            dispatch_key=dispatch_key,
+            dispatch_registrations_body=dispatch_registrations_body
+        )
+
     fm.write_with_template(f'Register{dispatch_key}.cpp', 'RegisterDispatchKey.cpp', lambda: {
+        'static_init_dispatch_registrations': static_init_dispatch_registrations,
+        'deferred_dispatch_registrations': deferred_dispatch_registrations,
```

Comment:
nit: you can probably just use one argument here instead of two in the template file (`dispatch_registrations`), since right now one of these two arguments is always an empty string.

Reply:
but I need to put the content in 2 different places, one place is inside anonymous namespace and the other is inside at:: namespace.  Maybe I'm missing something?

Reply:
ah yep, i forgot about that. makes sense!